packages/web-extension/src/observableWallet/util.ts

```diff
@@ -156,6 +156,7 @@ export const observableWalletProperties: RemoteApiProperties<ObservableWallet> =
     },
     rollback$: RemoteApiPropertyType.HotObservable
   },
+  unusedAddresses$: RemoteApiPropertyType.HotObservable,
```

Comment:
@mkazlauskas unusedAddresses$ is actually a cold observable, but looking at the code, tagging this as RemoteApiPropertyType.HotObservable will also work, let me know if there is any extra consideration here.

Reply:
Good point, indeed it's not going to work as expected, as messaging implementation [subscribes to underlying observable right away](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/web-extension/src/messaging/remoteApi.ts#L366-L367). So we have to either implement a new `RemoteApiPropertyType.ColdObservable`, or just go with a `MethodReturningPromise`. Happy to go either way since it's not trivial. Didn't realize that this would be an issue when I suggested to use an observable.

Reply:
Ok, if both options are equal, I just reverted the change of the observable (fixup commit dropped)